### ISSUE TITLE
Bridging is not active when doing solid fill over sparse fill

### ISSUE THREAD

**Issue by hippich** [alexrj/Slic3r#3146]
So I am playing with attached test model to test bridges mostly - https://thingiverse-production-new.s3.amazonaws.com/assets/b0/f3/01/ee/af/ultimate_calibration_test.STL

What I found that Slic3r uses regular solid fill pattern when filling over 5% infill. This results in broken solid infill, since: speed to to fast, often there is nothing to anchor to.

In my opinion, to get best adhesion, it should treat solid infill area as any other bridge. I.e. it first should do perimeter and then run bridge infill with appropriate speed/flow/width/etc.

![infill in the air](https://cloud.githubusercontent.com/assets/361630/11580225/b543ade2-99fa-11e5-9a51-98bfc5c418ca.jpg)

[ultimate_calibration_test-2015-12-3-19-48.gcode.txt](https://github.com/alexrj/Slic3r/files/51590/ultimate_calibration_test-2015-12-3-19-48.gcode.txt)

[ultimate_calibration_test.STL.txt](https://github.com/alexrj/Slic3r/files/51591/ultimate_calibration_test.STL.txt)

**Comment by mrvn** [comment]
I disagree on the speed. The speed ins't the problem. As I see it there are 2 solutions here:

1) extend the solid fill until the endpoint rests on something. That means either it hits a perimeter and the 15% overlap will hold the infill in place, or it hits the 5% infill and rests on that. For the infill cases it should extend a bit beyond the infill line so it rests solidly on it.

2) Create support lines in the layer(s) before. What I mean there is that it first should approximate the border of the solid fill that hangs in the air with a polygon(s) of few lines (approximate a circle with e.g. a larger octagon). Then it should extend each side of the polygon(s) till they rest on something and print those lines. Note: The lines would overlap at the corners of the polygon(s). The solid fill should then also be extended to rest on the polygon.

In both case the idea is to not have any endpoints hanging in the air and fall down. Option 1 is probably the best with high infill. Option 2 for sparse or concentric infill. Slic3r should probably do a mixture of the two. Try to find a supporting infill line near the solid fill border and create a support line when none is near.

**Comment by clough42** [comment]
I don't know if my situation is the same, but I'm seeing similar-looking failures, with unsupported extrusions ending in open space, like this:

![oddbridging1](https://cloud.githubusercontent.com/assets/3324188/12069760/6eda88a8-b007-11e5-9d95-5597b2506393.PNG)
![oddbridging2](https://cloud.githubusercontent.com/assets/3324188/12069762/7a070b2a-b007-11e5-8ad7-07dea2977168.PNG)
![oddbridging3](https://cloud.githubusercontent.com/assets/3324188/12069761/7a053d86-b007-11e5-9da3-bb9c5e2205b3.PNG)

Models and config files:

[ShroudBridges2.zip](https://github.com/alexrj/Slic3r/files/75855/ShroudBridges2.zip)

Sorry for the blast.  I'm attaching these examples to two or three open issues that look like they might be related instead of opening a new issue.
